File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
# 백준 1931
n = int(input())
arr = []
for i in range(n):
  s, e = map(int,input().split())
  arr.append([s,e])
arr.sort(key= lambda x:(x[1],x[0]))
#print(arr)

le = 0
answer = 0
for i in range(0,n):
  s,e = arr[i]
  if le <= s :
    le = e 
    answer+=1

print(answer)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.util.*;
import java.io.*;

public class Main{

	public static void main(String[] args) throws Exception{
		
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		int [][] mt = new int[n][2];
		int _max = 0;
		for(int i = 0 ; i < n ; i++) {
			String [] tokens = br.readLine().split(" ");
			mt[i][0] = Integer.parseInt(tokens[0]);
			mt[i][1] = Integer.parseInt(tokens[1]);
			_max = Math.max(_max, mt[i][1]);
		}
		Arrays.sort(mt,new Comparator<int[]>() {
			@Override
			public int compare(int[] a, int[] b) {
				return a[1]==b[1] ? a[0]-b[0] : a[1]-b[1];
			}
		});
		int answer = 0;
		int last = 0;
		for(int [] m : mt) {
			int start = m[0];
			int end = m[1];
			if(last <=start) {
				last = end;
				answer++;
			}
		}
		System.out.println(answer);
	}

}
